src/sentry/api/endpoints/relocations/index.py

```diff
@@ -165,37 +224,18 @@ def post(self, request: Request) -> Response:
 
         logger.info("relocations.index.post.start", extra={"caller": request.user.id})
 
-        try:
-            is_superuser = SuperuserPermission().has_permission(request, None)
-        except SuperuserRequired:
-            is_superuser = False
-        if not options.get("relocation.enabled") and not is_superuser:
-            return Response({"detail": ERR_FEATURE_DISABLED}, status=status.HTTP_403_FORBIDDEN)
-
         serializer = RelocationsPostSerializer(data=request.data)
         if not serializer.is_valid():
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
         validated = serializer.validated_data
         fileobj = validated.get("file")
+        file_size = fileobj.size
         owner_username = validated.get("owner")
         org_slugs = [org.strip() for org in validated.get("orgs").split(",")]
-        for org_slug in org_slugs:
-            if not re.match(ORG_SLUG_PATTERN, org_slug):
-                return Response(
-                    {"detail": ERR_INVALID_ORG_SLUG.substitute(org_slug=org_slug)},
-                    status=status.HTTP_400_BAD_REQUEST,
-                )
-
-        # Only superusers can start relocations for other users.
-        creator = user_service.get_user(user_id=request.user.id)
-        if creator is None:
-            raise RuntimeError("Could not ascertain request user's username")
-        if not is_superuser and creator.username != owner_username:
-            return Response(
-                {"detail": ERR_INVALID_OWNER.substitute(creator_username=creator.username)},
-                status=status.HTTP_400_BAD_REQUEST,
-            )
+        err = validate_new_relocation_request(request, owner_username, org_slugs, file_size)
+        if err is not None:
+            return err
 
         try:
```

Comment:
can this be moved up? then we can probably do something like below:
```
err = validate_new_relocation_request(
            request, owner.username, relocation.want_org_slugs, fileobj.size
        ) or validate_relocation_uniqueness(owner)
        if err is not None:
            return err
```

Reply:
No, we need this specific order. In your formulation, the first task we do would be to check for if the the owner exists, but we would have to do this before we check for permissions. This means that any user could use this endpoint to perform username sniffing ("does user foo exist? What about foo1? etc"), which is a security risk. Instead, we need to do `validate_new_relocation_request` first to ensure that the user has the correct permissions (they are either a superuser, OR they are not and the `owner` field matches their own username), and only then do we perform the owner check above.

If we combined the two checks like you suggest, we'd need to do lookup the owner first, which would break this ordering. We could ofc lookup the owner, cache the result, check for permissions, then report the owner error only if the permissions check passes, but that makes the code more difficult to read than the current setup imo,